### ISSUE TITLE
Cannot drag and drop scene root (and sometimes other nodes too) from Scene Tree to Inspector field because holding mouse button doesn't lock the inspector as it does with other nodes

### ISSUE THREAD

**Issue by hsandt** [godotengine/godot#78001]
### Godot version

v4.0.2.stable.official [7a0977ce2]

### System information

Linux Ubuntu 22.04 with Unity desktop

### Issue description

Every time I try to assign the root node to an inspector Assign... field, I lose the node focus so the Inspector changes and I cannot complete the drag-and-drop action.

![Godot 4 0 2 - Cannot drag root node (and sometimes other nodes too) without losing inspector focus](https://github.com/godotengine/godot/assets/1067772/9f1ab751-a21c-4325-888d-95c41b2d5f72)

~~It also sometimes happened with other nodes: when focusing root node and trying to drag a child node; or even when focus some child node and trying to drag another child node.~~

![Godot 4 0 2 - Cannot drag root node (and sometimes other nodes too) without losing inspector focus 2 (only in project)](https://github.com/godotengine/godot/assets/1067772/cbce0507-5724-4f86-96e4-f8daf0a74eb5)

Unfortunately, I could not reproduce the bug above in a fresh new project; it only happened in my full-fledged project.

=> So I'm removing this bug variant from this issue.

### Steps to reproduce

1. Open any MRP demo scene (there are several, but act the same).
2. Focus on a non-root node
3. Try to drag and drop the root node to any inspector field (there is often no real field to assign, but it doesn't matter; it's just to demonstrate inspector changing focus)

Expected: inspector should lock selection
Actual: inspector focuses on root node

And you can even try the variants to drag other nodes too, in your own projects, which may trigger the bug as more complex setup seem to trigger it more often (even if a brand new scene!)

### Minimal reproduction project

[v4.0.2 - Drag and drop another node sometimes doesnt lock inspector.zip](https://github.com/godotengine/godot/files/11687377/v4.0.2.-.Drag.and.drop.another.node.sometimes.doesnt.lock.inspector.zip)

**Comment by Sauermann** [comment]
Do you use X11 or Wayland?
Can't confirm issue on X11-Xfce v4.1.beta.custom_build [72b59325c].

**Comment by Sch1nken** [comment]
Can confirm the first (main) issue with the root-node being selected.

Second issue I could not replicate.

X11 (though this might only matter for the 2nd issue).

Tested on current master (a278c1b98a81738a35b96a933a6e6cf771f9ab2d).

**Comment by hsandt** [comment]
I use X11.

**Comment by hsandt** [comment]
> Second issue I could not replicate.

Could not replicate second issue (root focused, child node dragged) on v4.2.1.stable.official [b09f793f5]. Removing this sub-issue.

**Comment by dbnicholson** [comment]
It seems that the problem occurs [here](https://github.com/godotengine/godot/blob/af77100e394dcaca609b15bef815ed17475e51ed/editor/scene_tree_dock.cpp#L129-L130) in the SceneTreeDock. When clicking on the root item, the click is outside of the child SceneTreeEditor's rectangle. Since `tree_clicked` isn't set, the selection handler ends up pushing the target node immediately as the new edited node. I added some debug prints that you can see in this screencast (sorry the mouse pointer doesn't show up).

[godot-tree-selection.webm](https://github.com/user-attachments/assets/c0a7af1d-b5e8-4142-82b7-39859543e84f)

Here's the text from clicking on 2 children of the root node and then the root node:

```
scene tree rect: [P: (0, 50), S: (393, 501)], local mouse pos: (125, 79), has point: yes
scene tree rect: [P: (0, 50), S: (393, 501)], local mouse pos: (119, 114), has point: yes
scene tree rect: [P: (0, 50), S: (393, 501)], local mouse pos: (81, 15), has point: no
```

The SceneTreeEditor rectangle starts at y=50 below the root node. It seems like the SceneTreeEditor rectangle is wrong, but maybe this isn't the right way to check if the mouse click was inside a particular child node.

**Comment by hsandt** [comment]
Thanks for the PR!

Second issue (trying to drag child node to root node inspector filed) actually reappeared on 4.3.stable, but hopefully the PR above will fix both issues.

I'll try to compile code with this PR later to check if it solved second issue as well (or if I'm too slow, I'll see if second issue is fixed when 4.4 is released)
